src/pages/Settings/Settings.js

```diff
@@ -0,0 +1,173 @@
+import React, { useState } from 'react'
+import useForm from 'react-hook-form'
+import * as yup from 'yup'
+import Button from 'components/Button'
+
+import './Settings.module.css'
+
+export const PORT_NUMBER_REGEX = /^\d{4,6}$/
+export const EMAIL_REGEX = /^\S+@\S+$/i
+
+const SettingsValidationSchema = yup.object().shape({
+  hostName: yup.string().required(),
+  hostPubKey: yup.string().required(),
+  registrationEmail: yup.string()
+    .email()
+    .required(),
+  deviceName: yup.string().required(),
+  networkId: yup.string().required(),
+  deviceAdminPort: yup.number()
+    .min(1000)
+    .max(65000)
+    .required(),
+  hcAdminPort: yup.number()
+    .min(1000)
+    .max(65000)
+    .required(),
+  hcNetworkPort: yup.number()
+    .min(1000)
+    .max(65000)
+    .required(),
+  hostingPort: yup.number()
+    .min(1000)
+    .max(65000)
+    .required()
+})
+
+export function Settings ({
+  settings,
+  updateSettings,
+  factoryReset,
+  toggleSshAccess,
+  history: { push }
+}) {
+  // if (props.loading) return <h4>Loading Settings</h4>
+
+  const goToMenu = () => push('/menu')
+  const { register, handleSubmit, errors } = useForm({
+    defaultValues: settings,
+    validationSchema: SettingsValidationSchema
+  })
+
+  const [sshAccessVal, setSshAccess] = useState(false)
+
+  const handleViewTos = (e) => {
+    e.preventDefault()
+    push('/tos')
+  }
+
+  const handleToggleSshAccess = (e) => {
+    e.preventDefault()
+    setSshAccess(e.target.checked)
+    toggleSshAccess()
+  }
+
+  const onSubmit = values => {
+    // e.preventDefault()
+    console.log('!!! New settings:', values, errors)
+    updateSettings(settings)
+  }
+
+  console.log('!!! errors: ', errors)
```

Comment:
console

Reply:
👍 Leaving this on purpose as for Ugly UI sans data + new forms hook/API it's good to stay aware of it. I'll add a comment to this effect. Don't want to decide right now where we put these errors and will make sure C is including error states so we'll have a target.